src/class/video/video_device.c

```diff
@@ -1214,11 +1236,7 @@ uint16_t videod_open(uint8_t rhport, tusb_desc_interface_t const * itf_desc, uin
     if (0 == stm_idx && 1 == bInCollection) {
       /* If there is only one streaming interface and no alternate settings,
        * host may not issue set_interface so open the streaming interface here. */
-      uint8_t const *sbeg = (uint8_t const*)itf_desc + stm->desc.beg;
-      uint8_t const *send = (uint8_t const*)itf_desc + stm->desc.end;
-      if (end == _find_desc_itf(sbeg, send, _desc_itfnum(sbeg), 1)) {
-        TU_VERIFY(_open_vs_itf(rhport, stm, 0), 0);
-      }
+      TU_VERIFY(_open_vs_itf(rhport, stm, 0), 0);
```

Comment:
```suggestion
      uint8_t const *sbeg = (uint8_t const*)itf_desc + stm->desc.beg;
      uint8_t const *send = (uint8_t const*)itf_desc + stm->desc.end;
      if (send == _find_desc_itf(sbeg, send, _desc_itfnum(sbeg), 1)) {
        TU_VERIFY(_open_vs_itf(rhport, stm, 0), 0);
      }
```

I noticed a typo in the original code. It should be `send` instead of `end`. I apologize for the inconvenience. If you could change it to `send` to achieve the expected behavior, I would appreciate it.

Reply:
The modification addressed my previous concerns, and I've tested it thoroughly. It works very well.